src/App.tsx

```diff
@@ -1,39 +1,62 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.rightClickMouse);
+
+    document.addEventListener('click', this.leftClickMouse);
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
+    document.removeEventListener('contextmenu', this.rightClickMouse);
+    document.removeEventListener('click', this.leftClickMouse);
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  leftClickMouse = () => {
+    this.setState({ hasClock: true });
+  };
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  rightClickMouse = (event: MouseEvent) => {
+    event.preventDefault(); // not to show the context menu
 
-        {' time is '}
+    this.setState({ hasClock: false });
+  };
```

Comment:
3 rule in checklist
https://github.com/mate-academy/react_clock/blob/master/checklist.md